[16.0][IMP] purchase_vendor_promotion: auto assign supplier with promotion

trisdoan opened the pull request:
### This change
- In replenishment, we think that assigning best supplier should also be the intended logical behavior the module.

LoisRForgeFlow committed:
purchase_date_planned_manual

LoisRForgeFlow committed:
purchase_date_planned_manual: add tests

oca-transbot committed:
OCA Transbot updated translations from Transifex

MiquelRForgeFlow committed:
[MIG] purchase_date_planned_manual: Migration to 10.0

LoisRForgeFlow committed:
[11.0][MIG] purchase_date_planned_manual

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

MiquelRForgeFlow committed:
[IMP] Avoid merging lines with different date_planned

HviorForgeFlow committed:
[MIG] purchase_date_planned_manual: Migration to 12.0

MiquelRForgeFlow committed:
[IMP] Update predicted_arrival_late when po change state

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (6 of 6 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/zh_CN/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/pt_BR/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (6 of 6 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/sl/

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (6 of 6 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/pt_BR/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 16.6% (1 of 6 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_date_planned_manual/

ced-adhoc committed:
[FIX] purchase_request: changes in report layout

aliciagaarzo committed:
[16.0][IMP] purchase_discount: improve visual of purchase order line

ByteMeAsap committed:
[ADD] purchase_sign

pre-commit fixes

Test fixes

Removed unwanted check

Removed unwanted check for access

Reverted changes and updated test scripts

pre-commit fix

test fix

test fix

Test comment changes

Added tests

Removed commit

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_supplier_return
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_supplier_return/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (7 of 7 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_packaging_default
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_packaging_default/it/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (30 of 30 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_approved
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_approved/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (26 of 26 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_type
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_type/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (11 of 11 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_allowed_product
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_allowed_product/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (138 of 138 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_landed_cost
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_landed_cost/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (23 of 23 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_merge
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_merge/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (40 of 40 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_advance_payment
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_advance_payment/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (10 of 10 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_default_terms_conditions
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_default_terms_conditions/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (18 of 18 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_tag
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_tag/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (3 of 3 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-product_supplier_code_purchase/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (7 of 7 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_open_qty
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_open_qty/pt_BR/

chaule97 committed:
[IMP] purchase_date_planned_manual: pre-commit auto fixes

chaule97 committed:
[MIG] purchase_date_planned_manual: Migration to 16.0

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (11 of 11 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_fop_shipping
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_fop_shipping/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (13 of 13 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_discount
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_discount/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 29.4% (68 of 231 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_request
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_request/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (15 of 15 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_triple_discount
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_triple_discount/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (16 of 16 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-procurement_purchase_no_grouping
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-procurement_purchase_no_grouping/pt_BR/

carolinafernandez-tecnativa committed:
[16.0][ADD] product_supplierinfo_purchase_contact: Add Purchase contact in product supplier info

Commit:
[UPD] Update product_supplierinfo_purchase_contact.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-product_supplierinfo_purchase_contact
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-product_supplierinfo_purchase_contact/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (3 of 3 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-product_supplierinfo_purchase_contact
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-product_supplierinfo_purchase_contact/it/

legalsylvain committed:
[FIX] purchase_discount : transmits correctly discount value to account.move.line, if account_invoice_triple_discount is installed. (See refactoring https://github.com/OCA/account-invoicing/pull/1638)

trisdoan committed:
[IMP] purchase_vendor_promotion: auto assign supplier with promotion

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update purchase_date_planned_manual.pot

OCA-git-bot committed:
[BOT] post-merge updates

santostelmo commented on purchase_vendor_promotion/models/stock_warehouse_orderpoint.py:
```diff
@@ -12,6 +12,24 @@ class StockWarehouseOrderpoint(models.Model):
+                    lambda x: x._is_promotion_active_or_upcoming()
+                )
+                if promotion_suppliers:
+                    rec.supplier_id = promotion_suppliers.sorted(key=lambda x: x.price)[
```
Do you need to sort again by price ?  That was already done in `_prepare_sellers` no ?

trisdoan replied:
Hello @santostelmo, you're totally right. I wanted to prioritize the price at the beginning. But I think it's not necessary to repeat it.

Commit:
[UPD] Update purchase_sign.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_date_planned_manual
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_date_planned_manual/it/

Commit:
[UPD] Update purchase_vendor_promotion.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_vendor_promotion
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_vendor_promotion/it/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (11 of 11 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_allowed_product
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_allowed_product/sv/

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (26 of 26 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_type
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_type/sv/

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (231 of 231 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_request
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_request/sv/

trisdoan commented:
Hello @santostelmo , @cyrilmanuel , I prepared https://github.com/OCA/purchase-workflow/pull/2321 in order to include a multi-company issue fix :pray:
